gh: phase out testing with kernel-of-the-day LVH images

julianwiedmann opened the pull request:
Testing unstable Cilium changes (from PRs) against unstable infrastructure (random kernel-of-the-day builds) doesn't really give us much information. Switch to the latest LTS kernel instead.

Keeping bpf-next in the datapath verifier workflow (it's a good sniff test for complexity changes).

If we still want to sniff-test against bleeding-edge upstream kernels, then this could be done with scheduled runs. But not in PRs.

julianwiedmann committed:
gh: phase out testing with kernel-of-the-day LVH images

Testing unstable Cilium changes (from PRs) against unstable infrastructure
(random kernel-of-the-day builds) doesn't really give us much information.
Switch to the latest LTS kernel instead.

Keeping bpf-next in the datapath verifier workflow (it's a good sniff test
for complexity changes), and in ginkgo (the kernel name is hard-baked into
the test infra).

If we still want to sniff-test against bleeding-edge upstream kernels,
then this could be done with scheduled runs. But not in PRs.

Signed-off-by: Julian Wiedmann <jwi@isovalent.com>

aanm reviewed (approved):
:ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship: :ship:

julianwiedmann committed:
gh: ginkgo: phase out testing with bpf-next kernel

Switch to the latest LTS kernel, but pretend that it's NETNEXT for all
the ginkgo-internal logic.

Signed-off-by: Julian Wiedmann <jwi@isovalent.com>

aanm commented on .github/actions/ginkgo/main-k8s-versions.yaml:
```diff
@@ -6,25 +6,29 @@ include:
     kube-image: "quay.io/cilium/kindest-node:v1.31.0@sha256:d2b2a8cd6fa282b9a4126938341a4d2924dfa96f60b1f983d519498c9cde1a99"
     # renovate: datasource=docker depName=quay.io/lvh-images/kind
     kernel: "rhel8.6-20250319.212144"
+    kernel-type: "oldstable"
```
It would be nice to leave a comment of what does "oldstable" vs "latest" vs "stable" means.

julianwiedmann replied:
I would if I knew how we initially picked those kernels :sweat_smile:

tklauser commented on .github/workflows/conformance-runtime.yaml:
```diff
@@ -266,7 +266,7 @@ jobs:
           install-dependencies: true
           # renovate: datasource=docker depName=quay.io/lvh-images/kind
-          image-version: "bpf-next-20250110.013326"
+          image-version: "6.12-20241218.004849"
```
Along with this change, I think we should probably remove

https://github.com/cilium/cilium/blob/ccb5a1a2e6a47e5068d7058cd05d145c56590e9c/.github/renovate.json5#L180-L189

so the image will be updated by renovate again going forward.

julianwiedmann replied:
I remembered that https://github.com/cilium/cilium/issues/38058 was also affecting stable branches (with a stable-version kernel), so I think we need to keep this renovate config.
